addons/mrp_account/models/mrp_production.py

```diff
@@ -41,6 +41,9 @@ def write(self, vals):
                 for workorder in production.workorder_ids:
                     workorder.mo_analytic_account_line_id.ref = production.display_name
                     workorder.mo_analytic_account_line_id.name = _("[WC] %s", workorder.display_name)
+        if vals.get('analytic_account_id'):
+            for production in self:
+                production.move_raw_ids.analytic_account_line_id.account_id = vals['analytic_account_id']
```

Comment:
`vals['analytic_account_id']` is a integer you can't use it directly
"For loop" is useless in this case, you can assign in batch

Reply:
Thanks for the review!
Could you please check the update

Reply:
Forget about the integer thing, apparently you can assign it directly. My bad.
Another thing I noticed is when change `analytic_account_id` to `False`, you should unlink all the lines.